language/move-vm/runtime/src/interpreter.rs

```diff
@@ -1019,11 +1238,21 @@ impl Frame {
                         let reference = interpreter.operand_stack.pop_as::<Reference>()?;
                         gas_meter.charge_read_ref(reference.value_view())?;
                         let value = reference.read_ref()?;
+
+                        if self.runtime_config.paranoid_hot_potato_checks {
+                            value.has_ability(Ability::Copy)?;
+                        }
+
                         interpreter.operand_stack.push(value)?;
                     }
                     Bytecode::WriteRef => {
                         let reference = interpreter.operand_stack.pop_as::<Reference>()?;
                         let value = interpreter.operand_stack.pop()?;
+
+                        if self.runtime_config.paranoid_hot_potato_checks {
```

Comment:
```suggestion
                        if self.runtime_config.paranoid_type_checks {
```

Reply:
This bug probably also means we're missing a test that exercises this case when `paranoid_hot_potato_checks` is off, but `paranoid_type_checks` is on

Reply:
I think I'll just combine the two flags. The original idea is we have `paranoid_type_checks ` for type safety and `paranoid_hot_potato_checks ` for ability safety. However, `paranoid_hot_potato_checks ` won't work unless we turn `paranoid_type_checks` on because we need value tagging..